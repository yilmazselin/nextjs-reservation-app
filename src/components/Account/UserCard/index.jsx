import { useTranslation } from "next-i18next"
import React, { memo } from "react"
import styled from "styled-components"
import Text from "../../common/Text"
import { colors, fontSize, fontWeight, padding } from "../../../utils/styles"

const Container = styled.div`
  border-bottom: 1px solid ${colors.light};
  padding-bottom: ${padding.xs}px;
  margin-bottom: ${padding.md}px;
`
const textStyle = {
  fontSize: fontSize.nm,
  color: colors.dark,
  fontWeight: fontWeight.semiBold,
  params: {
    letterSpacing: "0.4px",
    marginBottom: padding.sm
  }
}

function UserCard({ currentLocale, data }) {
  const { t } = useTranslation("account")

  return (
    <Container>
      <Text {...textStyle}>
        {t("account:email")}: {data?.user.email}
      </Text>
      <Text {...textStyle}>
        {t("account:password")}: {data?.user.password}
      </Text>
      <Text {...textStyle}>
        {t("account:locale")}: {currentLocale}
      </Text>
    </Container>
  )
}

export default memo(UserCard)
