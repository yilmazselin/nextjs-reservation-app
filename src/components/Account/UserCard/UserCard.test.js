import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import "jest-styled-components"
import { useTranslation } from "react-i18next"
import UserCard from "."

jest.mock("react-i18next", () => ({
  useTranslation: jest.fn()
}))
const tSpy = jest.fn((str) => str)

useTranslation.mockReturnValue({
  t: tSpy,
  i18n: {
    changeLanguage: () => new Promise(() => {})
  }
})

describe("<UserCard />", () => {
  it("Should render correctly", () => {
    const { container } = render(<UserCard />)

    expect(container).toMatchSnapshot()
  })

  it("Should viewed the user data", () => {
    const data = {
      user: {
        email: "test@gmail.com",
        password: "1234",
        locale: "tr"
      }
    }

    render(<UserCard currentLocale={data.user.locale} data={data} />)

    expect(screen.getByText(/account:email: test@gmail.com/i)).toBeInTheDocument()
    expect(screen.getByText(/account:password: 1234/i)).toBeInTheDocument()
    expect(screen.getByText(/account:locale: tr/i)).toBeInTheDocument()
  })
})
