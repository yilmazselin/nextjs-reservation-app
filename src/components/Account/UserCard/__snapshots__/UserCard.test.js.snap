// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCard /> Should render correctly 1`] = `
.c1 {
  color: #121212;
  font-size: 14px;
  font-family: "NunitoSans-600","sans-serif";
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  margin-bottom: 16px;
}

.c0 {
  border-bottom: 1px solid #E6E6E6;
  padding-bottom: 8px;
  margin-bottom: 24px;
}

<div>
  <div
    class="c0"
  >
    <p
      class="c1"
      color="#121212"
      font-size="14"
      font-weight="600"
    >
      account:email
      : 
    </p>
    <p
      class="c1"
      color="#121212"
      font-size="14"
      font-weight="600"
    >
      account:password
      : 
    </p>
    <p
      class="c1"
      color="#121212"
      font-size="14"
      font-weight="600"
    >
      account:locale
      : 
    </p>
  </div>
</div>
`;
